## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 403264`
Commit: `8467d0a1542665a293e15c16f4241edf9b81b67d`

---
### `x86_64-linux`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>charmcraft</li>
    <li>charmcraft.dist</li>
    <li>rockcraft</li>
    <li>rockcraft.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 18 packages built:</summary>
  <ul>
    <li>python312Packages.craft-application</li>
    <li>python312Packages.craft-application.dist</li>
    <li>python312Packages.craft-parts</li>
    <li>python312Packages.craft-parts.dist</li>
    <li>python312Packages.craft-providers</li>
    <li>python312Packages.craft-providers.dist</li>
    <li>python312Packages.logassert</li>
    <li>python312Packages.logassert.dist</li>
    <li>python313Packages.craft-application</li>
    <li>python313Packages.craft-application.dist</li>
    <li>python313Packages.craft-parts</li>
    <li>python313Packages.craft-parts.dist</li>
    <li>python313Packages.craft-providers</li>
    <li>python313Packages.craft-providers.dist</li>
    <li>python313Packages.logassert</li>
    <li>python313Packages.logassert.dist</li>
    <li>snapcraft</li>
    <li>snapcraft.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>rockcraft</summary>
<pre>Cannot create packages outside of the project tree.
Detailed information: The following file paths are invalid:
- /build/pytest-of-nixbld/pytest-0/test_run_pack_services0/my-rock.rock
Recommended resolution: Change the output directory when packing.
---------------------------- Captured log teardown -----------------------------
WARNING  root:features.py:42 deleting current features configuration
=============================== warnings summary ===============================
tests/unit/test_oci.py:39
  /build/source/tests/unit/test_oci.py:39: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
    f"{(datetime.datetime.utcnow() - datetime.datetime(1970, 1, 1)).days}"

tests/unit/test_oci.py::TestImage::test_append_new_user[base_user_files0-prime_user_files0-True-expected_user_files0]
tests/unit/test_oci.py::TestImage::test_append_new_user[base_user_files1-prime_user_files1-False-expected_user_files1]
  /build/source/rockcraft/oci.py:287: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
    days_since_epoch = (datetime.utcnow() - datetime(1970, 1, 1)).days  # type: ignore[reportDeprecated]

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/unit/test_cli.py::test_run_pack_services - assert False
====== 1 failed, 402 passed, 8 skipped, 3 deselected, 3 warnings in 1.20s ======</pre>
</details>
</details>

--- a/pkgs/development/python-modules/craft-application/default.nix
+++ b/pkgs/development/python-modules/craft-application/default.nix
@@ -10,7 +10,7 @@
   craft-providers,
   jinja2,
   fetchFromGitHub,
-  git,
+  gitMinimal,
   hypothesis,
   license-expression,
   nix-update-script,
@@ -20,29 +20,33 @@
   pytest-mock,
   pytest-subprocess,
   pytestCheckHook,
-  pythonOlder,
   pyyaml,
   responses,
   setuptools-scm,
   snap-helpers,
   freezegun,
+  cacert,
+  writableTmpDirAsHomeHook,
 }:
 
 buildPythonPackage rec {
   pname = "craft-application";
-  version = "4.10.0";
+  version = "5.3.0";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "canonical";
     repo = "craft-application";
     tag = version;
-    hash = "sha256-9M49/XQuWwKuQqseleTeZYcrwd/S16lNCljvlVsoXbs=";
+    hash = "sha256-6iD35ql3/vUzILh5VMWiFwBKPoGPfCUgEKD4g7s55Y0=";
   };
 
   postPatch = ''
     substituteInPlace pyproject.toml \
       --replace-fail "setuptools==75.8.0" "setuptools"
+
+      substituteInPlace craft_application/git/_utils.py \
+        --replace-fail "/snap/core22/current/etc/ssl/certs" "${cacert}/etc/ssl/certs"
   '';
 
   build-system = [ setuptools-scm ];
@@ -68,7 +72,7 @@ buildPythonPackage rec {
 
   nativeCheckInputs = [
     freezegun
-    git
+    gitMinimal
     hypothesis
     pyfakefs
     pytest-check
@@ -76,11 +80,10 @@ buildPythonPackage rec {
     pytest-subprocess
     pytestCheckHook
     responses
+    writableTmpDirAsHomeHook
   ];
 
   preCheck = ''
-    export HOME=$(mktemp -d)
-
     # Tests require access to /etc/os-release, which isn't accessible in
     # the test environment, so create a fake file, and modify the code
     # to look for it.
@@ -90,6 +93,11 @@ buildPythonPackage rec {
 
     substituteInPlace craft_application/util/platforms.py \
       --replace-fail "os_utils.OsRelease()" "os_utils.OsRelease(os_release_file='$HOME/os-release')"
+
+    # Not using `--replace-fail` here only because it simplifies overriding this package in the charmcraft
+    # derivation. Once charmcraft has moved to craft-application >= 5, `--replace-fail` can be added.
+    substituteInPlace tests/conftest.py \
+      --replace "include_lsb=False, include_uname=False, include_oslevel=False" "include_lsb=False, include_uname=False, include_oslevel=False, os_release_file='$HOME/os-release'"
   '';
 
   pythonImportsCheck = [ "craft_application" ];
@@ -109,6 +117,10 @@ buildPythonPackage rec {
       # slightly in a later revision of craft-platforms. No functional error.
       "test_platform_invalid_arch"
       "test_platform_invalid_build_arch"
+      # Asserts against string output which fails when not on Ubuntu.
+      "test_run_error_with_docs_url"
+      # Asserts a fallback path for SSL certs that we override in a patch.
+      "test_import_fallback_wrong_metadata"
     ]
     ++ lib.optionals stdenv.hostPlatform.isAarch64 [
       # These tests have hardcoded "amd64" strings which fail on aarch64
@@ -116,6 +128,11 @@ buildPythonPackage rec {
       "test_process_grammar_platform"
       "test_process_grammar_default"
     ];
+
+  disabledTestPaths = [
+    # These tests assert outputs of commands that assume Ubuntu-related output.
+    "tests/unit/services/test_lifecycle.py"
+  ];
 
   passthru.updateScript = nix-update-script { };
 

--- a/pkgs/development/python-modules/craft-parts/default.nix
+++ b/pkgs/development/python-modules/craft-parts/default.nix
@@ -25,11 +25,12 @@
   ant,
   maven,
   jdk,
+  writableTmpDirAsHomeHook,
 }:
 
 buildPythonPackage rec {
   pname = "craft-parts";
-  version = "2.8.0";
+  version = "2.10.0";
 
   pyproject = true;
 
@@ -37,7 +38,7 @@ buildPythonPackage rec {
     owner = "canonical";
     repo = "craft-parts";
     tag = version;
-    hash = "sha256-1SnT/yB6vJm82yhszBRjeph13B91KYap8/KR4L9VcjM=";
+    hash = "sha256-nGOLzQ+mlLHmqkknWklCIEN7HOgQznuYO1rbqUvL1N4=";
   };
 
   patches = [ ./bash-path.patch ];
@@ -76,13 +77,10 @@ buildPythonPackage rec {
     requests-mock
     socat
     squashfsTools
+    writableTmpDirAsHomeHook
   ];
 
   pytestFlagsArray = [ "tests/unit" ];
-
-  preCheck = ''
-    export HOME=$(mktemp -d)
-  '';
 
   disabledTests = [
     # Relies upon paths not present in Nix (like /bin/bash)

--- a/pkgs/development/python-modules/craft-platforms/default.nix
+++ b/pkgs/development/python-modules/craft-platforms/default.nix
@@ -14,7 +14,7 @@
 
 buildPythonPackage rec {
   pname = "craft-platforms";
-  version = "0.8.0";
+  version = "0.9.0";
   pyproject = true;
 
   disabled = pythonOlder "3.10";
@@ -23,7 +23,7 @@ buildPythonPackage rec {
     owner = "canonical";
     repo = "craft-platforms";
     tag = version;
-    hash = "sha256-U57hmQ3UPuwoue8kAxAXiH8ecViryFqIxmpnaAdQnZo=";
+    hash = "sha256-Ec4zncigqampXND0YvDnExrrB9ls3Sf0KfYVbwLGbqY=";
   };
 
   postPatch = ''

